src/Polly.Core/Simmy/Outcomes/OutcomeChaosStrategy.cs

```diff
@@ -0,0 +1,119 @@
+{
+    private readonly ResilienceStrategyTelemetry _telemetry;
+
+    public OutcomeChaosStrategy(OutcomeStrategyOptions<Exception> options, ResilienceStrategyTelemetry telemetry)
+        : base(options)
+    {
+        if (options.Outcome is null && options.OutcomeGenerator is null)
+        {
+            throw new ArgumentNullException(nameof(options.Outcome), "Either Outcome or OutcomeGenerator is required.");
+        }
+
+        _telemetry = telemetry;
+        Fault = options.Outcome;
+        OnFaultInjected = options.OnOutcomeInjected;
+        FaultGenerator = options.OutcomeGenerator is not null ? options.OutcomeGenerator : (_) => new(options.Outcome);
+    }
+
+    public OutcomeChaosStrategy(OutcomeStrategyOptions<T> options, ResilienceStrategyTelemetry telemetry)
+        : base(options)
+    {
+        if (options.Outcome is null && options.OutcomeGenerator is null)
+        {
+            throw new ArgumentNullException(nameof(options.Outcome), "Either Outcome or OutcomeGenerator is required.");
+        }
+
+        _telemetry = telemetry;
+        Outcome = options.Outcome;
+        OnOutcomeInjected = options.OnOutcomeInjected;
+        OutcomeGenerator = options.OutcomeGenerator is not null ? options.OutcomeGenerator : (_) => new(options.Outcome);
+    }
+
+    public Func<OnOutcomeInjectedArguments<T>, ValueTask>? OnOutcomeInjected { get; }
+
+    public Func<OnOutcomeInjectedArguments<Exception>, ValueTask>? OnFaultInjected { get; }
+
+    public Func<OutcomeGeneratorArguments, ValueTask<Outcome<T>?>>? OutcomeGenerator { get; }
+
+    public Func<OutcomeGeneratorArguments, ValueTask<Outcome<Exception>?>>? FaultGenerator { get; }
+
+    public Outcome<T>? Outcome { get; }
+
+    public Outcome<Exception>? Fault { get; }
+
+    protected internal override async ValueTask<Outcome<T>> ExecuteCore<TState>(Func<ResilienceContext, TState, ValueTask<Outcome<T>>> callback, ResilienceContext context, TState state)
+    {
+        try
+        {
+            if (await ShouldInjectAsync(context).ConfigureAwait(context.ContinueOnCapturedContext))
+            {
+                if (FaultGenerator is not null)
+                {
+                    var fault = await InjectFault(context).ConfigureAwait(context.ContinueOnCapturedContext);
+                    if (fault is not null)
+                    {
+                        return new Outcome<T>(fault);
+                    }
+                }
+                else if (OutcomeGenerator is not null)
+                {
+                    var outcome = await InjectOutcome(context).ConfigureAwait(context.ContinueOnCapturedContext);
+                    if (outcome.HasValue)
+                    {
+                        return new Outcome<T>(outcome.Value.Result);
+                    }
+                }
+                else
+                {
+                    throw new InvalidOperationException("Either a fault or fake outcome to inject must be defined.");
+                }
```

Comment:
Should we just validate this in the constructor once?

Reply:
Removed it since we're already doing so in the ctor 👍